pxemgr/ipxe_handlers.go

```diff
@@ -696,3 +698,31 @@ func (mgr *pxeManagerT) hostCoreOSVersion(r *http.Request) string {
 
 	return coreOSversion
 }
+
+func (mgr *pxeManagerT) getMayuConfig(w http.ResponseWriter, r *http.Request) {
+	confBytes, err := yaml.Marshal(mgr.config)
+	if err != nil {
+		return
+	}
+	io.WriteString(w, string(confBytes))
+}
+
+func (mgr *pxeManagerT) setMayuConfig(w http.ResponseWriter, r *http.Request) {
+	var conf Configuration
+
+	data, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		panic(err)
+	}
+	defer r.Body.Close()
+
+	err = yaml.Unmarshal(data, &conf)
+	if err != nil {
+		panic(err)
+		return
```

Comment:
The return is not required if you panic. In the long run I would like to do proper logging. In our microservices we decided to do structured logging. We also agreed to prevent using `panic` whenever possible.